lib/SPIRV/SPIRVInternal.h

```diff
@@ -376,6 +376,7 @@ const static char WGSize[] = "reqd_work_group_size";
 const static char WGSizeHint[] = "work_group_size_hint";
 const static char SubgroupSize[] = "intel_reqd_sub_group_size";
 const static char MaxWGSize[] = "max_work_group_size";
+const static char NoGlobOffset[] = "uses_global_work_offset";
```

Comment:
I'm confused with the naming. Should the metadata name be something like `no_global_work_offset` ?

Reply:
Agree, I will rename this metadata variable name to `UsesGlobalOffset` to avoid this confusing difference

Reply:
The name of the metadata that the FPGA compiler expecting is **uses_global_work_offset**.  Are you proposing to change that?

Reply:
The way you are proposing it is confusing even more. Why the variable should be called `UsesGlobalOffset` while the execution mode in the spec is called `NoGlobalOffsetINTEL`?
Just looking at `UsesGlobalOffset` name makes me think that we **do** use global offset in some meaningful way, but description of `NoGlobalOffsetINTEL` tells me that the offset is 0, which is effectively means we **do not** use offset.

Reply:
Here the variable name should correlate with LLVM IR metadata naming, so it is called `UsesGlobalOffset` (see the last changes in PR).

I see that all the rest can be confusing, will try to describe it in details:
`uses_global_work_offset` metadata instructs the compiler **to use** or **not to use** the global work offset values. Its valid values are `0` and `1`, where `0` means that we **do not** use the global offset. And this case is mapped/translated to `NoGlobalOffsetINTEL` execution mode in SPIR-V. This logic is implemented in `PreprocessMetadata.cpp`

Reply:
> The name of the metadata that the FPGA compiler expecting is **uses_global_work_offset**. Are you proposing to change that?

Yes indeed. Is it possible, @mendell27?

Reply:
Please do **not** change the name.